inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/consume/InlongConsumeServiceImpl.java

```diff
@@ -0,0 +1,165 @@
+import org.apache.inlong.manager.pojo.consumption.InlongConsumeRequest;
+import org.apache.inlong.manager.pojo.user.UserRoleCode;
+import org.apache.inlong.manager.service.user.LoginUserUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.CollectionUtils;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+
+/**
+ * Inlong consume service layer implementation
+ */
+@Slf4j
+@Service
+public class InlongConsumeServiceImpl implements InlongConsumeService {
+
+    @Autowired
+    private InlongConsumeEntityMapper consumeEntityMapper;
+
+    @Autowired
+    private InlongConsumeOperatorFactory consumeOperatorFactory;
+
+    @Override
+    public Integer save(InlongConsumeRequest consumeRequest, String operator) {
+        log.debug("begin to save consumption info={}", consumeRequest);
+        Preconditions.checkNotNull(consumeRequest, "consumption info cannot be null");
+        Preconditions.checkNotNull(consumeRequest.getTopic(), "consumption topic cannot be empty");
+        if (isConsumerGroupExists(consumeRequest.getConsumerGroup(), consumeRequest.getId())) {
+            throw new BusinessException(String.format("consumer group %s already exist",
+                    consumeRequest.getConsumerGroup()));
+        }
+
+        if (consumeRequest.getId() != null) {
+            InlongConsumeEntity consumeEntity = consumeEntityMapper.selectByPrimaryKey(consumeRequest.getId());
+            Preconditions.checkNotNull(consumeEntity, "consumption not exist with id: " + consumeRequest.getId());
+            ConsumptionStatus consumptionStatus = ConsumptionStatus.fromStatus(consumeEntity.getStatus());
+            Preconditions.checkTrue(ConsumptionStatus.ALLOW_SAVE_UPDATE_STATUS.contains(consumptionStatus),
+                    "consumption not allow update when status is " + consumptionStatus.name());
+        }
+
+        InlongConsumeOperator instance = consumeOperatorFactory.getInstance(consumeRequest.getMqType());
+        instance.setTopicInfo(consumeRequest);
+        instance.saveOpt(consumeRequest, operator);
+        return consumeRequest.getId();
+    }
+
+    @Override
+    public boolean isConsumerGroupExists(String consumerGroup, Integer excludeSelfId) {
+        ConsumptionQuery consumptionQuery = new ConsumptionQuery();
+        consumptionQuery.setConsumerGroup(consumerGroup);
+        consumptionQuery.setIsAdminRole(true);
+        List<InlongConsumeEntity> result = consumeEntityMapper.listByQuery(consumptionQuery);
+        if (excludeSelfId != null) {
+            result = result.stream().filter(consumer -> !excludeSelfId.equals(consumer.getId()))
+                    .collect(Collectors.toList());
+        }
+        return !CollectionUtils.isEmpty(result);
```

Comment:
Could use apache.commons.collectionUtils.CollectionUtils.isNotEmpty()
Easier to understand